packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerBuilder.java

```diff
@@ -40,108 +40,143 @@ private static Paint setTextPaint(float size, Context context) {
         return paint;
     }
 
-    @NonNull
-    private static Paint getBackgroundColor() {
+    private Paint getMarkerPaint() {
         Paint paint = new Paint();
         paint.setColor(Color.WHITE);
         paint.setAntiAlias(true);
         return paint;
     }
 
-    @NonNull
-    private static Paint getShadowPaint() {
+    private Paint getShadowPaint() {
         Paint paint = new Paint();
         paint.setColor(Color.BLACK);
         paint.setStyle(Paint.Style.STROKE);
         paint.setAlpha(15);
-        paint.setStrokeWidth(6);
-        paint.setMaskFilter(new BlurMaskFilter(8, BlurMaskFilter.Blur.NORMAL));
+        paint.setStrokeWidth(shadowSize);
         paint.setAntiAlias(true);
         return paint;
     }
 
-    private static int getMarkerSize() {
+    private int getMarkerSize() {
         int baseScreenHeight = 2467;
         int baseMarkerSize = 167;
         int maxMarkerSize = 172;
         int minMarkerSize = 67;
 
         int physicalPixelHeight = Resources.getSystem().getDisplayMetrics().heightPixels;
-        double heightRatio = ((double ) (physicalPixelHeight)) / ((double) (baseScreenHeight));
+        double heightRatio = ((double) (physicalPixelHeight)) / ((double) (baseScreenHeight));
         int proportionalMarkerSize = (int) (baseMarkerSize * heightRatio);
 
         if (proportionalMarkerSize > maxMarkerSize) {
             return maxMarkerSize;
-        } else if (proportionalMarkerSize < minMarkerSize) {
-            return minMarkerSize;
-        } else {
-            return proportionalMarkerSize;
-        }
+        } else
+            return Math.max(proportionalMarkerSize, minMarkerSize);
     }
 
-    private static Bitmap getClusterBitmap(int size) {
+    private int getBorderRadius(int multiply) {
+        double density = Resources.getSystem().getDisplayMetrics().density;
+        return (int) (density * multiply);
+    }
+
+    private Bitmap getEmptyClusterBitmap(int size) {
         Bitmap marker = Bitmap.createBitmap(size, size, Bitmap.Config.ARGB_8888);
         Canvas canvas = new Canvas(marker);
         canvas.drawCircle(size / 2f, size / 2f, size / 2.2f, getShadowPaint());
-        canvas.drawCircle(size / 2f, size / 2f, size / 2.2f, getBackgroundColor());
+        canvas.drawCircle(size / 2f, size / 2f, (size / 2.2f) - shadowSize, getMarkerPaint());
```

Comment:
Here i'm subtracting the shadowSize to make the actual cluster smaller than the shadow.